File: global/global-workflow-boot-starter/src/main/java/com/particle/global/workflow/dto/WorkflowData.java
package com.particle.global.workflow.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import java.util.List;

/**
 * 工作流定义数据 — 节点和边的集合
 * <p>
 * 对应前端 WorkflowGraphData.definition（VueFlowGraphData 的 nodes + edges 部分）。
 * 不包含 viewport 等前端专属字段。
 * </p>
 *
 * @author particle
 * @since 2026-05-03
 */
@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class WorkflowData {

    /**
     * 节点列表
     */
    private List<WorkflowNode> nodes;

    /**
     * 边列表
     */
    private List<WorkflowEdge> edges;
}

File: web/global/pc/common/vueflowWorkflow/workflow/WorkflowData.ts
import type { WorkflowNodeData } from './WorkflowNodeData'
import type { WorkflowEdgeData } from './WorkflowEdgeData'
// ==================== 工作流数据结构 ====================
/**
 * 工作流数据
 * 对应后端的工作流数据
 */
export interface WorkflowData {
    nodes: WorkflowNode[];
    edges: WorkflowEdge[];
}

/**
 * 定义节点
 * 对应后端的工作流节点数据
 */
export interface WorkflowNode {
    id: string;
    data: WorkflowNodeData;
}

/**
 * 定义边
 * 对应后端的工作流边数据
 */
export interface WorkflowEdge {
    id: string;
    data: WorkflowEdgeData;
}
